app/controllers/schemes_controller.rb

```diff
@@ -170,14 +174,28 @@ def scheme_params
                                                      :intended_stay,
                                                      :confirmed)
 
-    full_params = required_params[:arrangement_type] == "D" && required_params[:owning_organisation_id].present? ? required_params.merge(managing_organisation_id: required_params[:owning_organisation_id]) : required_params
+    if arrangement_type_changed_to_different_org?(required_params)
+      required_params[:managing_organisation_id] = nil
+    end
+
+    if arrangement_type_set_to_same_org?(required_params)
+      required_params[:managing_organisation_id] = required_params[:owning_organisation_id] || @scheme.owning_organisation_id
+    end
 
-    full_params[:sensitive] = full_params[:sensitive].to_i if full_params[:sensitive]
+    required_params[:sensitive] = required_params[:sensitive].to_i if required_params[:sensitive]
 
     if current_user.data_coordinator?
-      full_params[:owning_organisation_id] = current_user.organisation_id
+      required_params[:owning_organisation_id] = current_user.organisation_id
     end
-    full_params
+    required_params
+  end
+
+  def arrangement_type_set_to_same_org?(required_params)
+    required_params[:arrangement_type] == "The same organisation that owns the housing stock" || (required_params[:arrangement_type].blank? && @scheme.present? && @scheme.arrangement_type_same?)
```

Comment:
Be good to remove the string comparison here too

Reply:
The only thing I can think of doing is using the map to get the db value, but if we're thinking of refactoring this to pass the value in the params at some point, would that still be useful?

Reply:
I think so yes.

A) Because I'm not sure we'll ever actually do that refactor, and B) because even if we do, I don't think it makes that refactor any harder and makes this easier to change in the meantime?